pkg/apis/storage/types.go

```diff
@@ -145,15 +145,22 @@ type VolumeAttachmentSpec struct {
 }
 
 // VolumeAttachmentSource represents a volume that should be attached.
-// Right now only PersistenVolumes can be attached via external attacher,
-// in future we may allow also inline volumes in pods.
+// Right now PersistenVolumes as well as inline volumes (only in
+// CSI Migration scenarios) can be attached via external attacher.
 // Exactly one member can be set.
 type VolumeAttachmentSource struct {
 	// Name of the persistent volume to attach.
 	// +optional
 	PersistentVolumeName *string
 
-	// Placeholder for *VolumeSource to accommodate inline volumes in pods.
+	// Contains all the information necessary to attach a persistent
+	// volume defined by a pod's inline VolumeSource. This field is
+	// populated only for the CSIMigration feature. It contains
+	// translated fields from a pod's inline VolumeSource to a
+	// PersistentVolumeSpec. This field is alpha-level and is only
+	// honored by servers that enabled the CSIMigration feature.
+	// +optional
+	InlineVolumeSpec *api.PersistentVolumeSpec
```

Comment:
It would be great to have a comment (not part of generated docs?) why `PersistentVolumeSpec` is used instead of `CSIPersistentVolumeSource` - we need at least read-only flag translated into AccessMode and maybe MountOptions.

Reply:
Addressed this in the usage of this field in CSI attacher code.